File: app_mobile/gest_squadre/lib/screens/launch_splash_screen.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';

import '../widgets/logo_badge.dart';

/// Splash iniziale con motion come TOC (`_ForcedLaunchSplashScreen`).
class LaunchSplashScreen extends StatefulWidget {
  const LaunchSplashScreen({super.key});

  @override
  State<LaunchSplashScreen> createState() => _LaunchSplashScreenState();
}

class _LaunchSplashScreenState extends State<LaunchSplashScreen>
    with SingleTickerProviderStateMixin {
  late final AnimationController _splashController;
  late final Animation<double> _logoScale;
  late final Animation<double> _logoAlignY;
  late final Animation<double> _titleOpacity;
  late final Animation<double> _signatureOpacity;

  @override
  void initState() {
    super.initState();
    _splashController = AnimationController(
      vsync: this,
      duration: const Duration(milliseconds: 4300),
    );
    _logoScale = Tween<double>(begin: 0.28, end: 1.0).animate(
      CurvedAnimation(parent: _splashController, curve: Curves.easeOutCubic),
    );
    _logoAlignY = Tween<double>(begin: -0.78, end: -0.22).animate(
      CurvedAnimation(parent: _splashController, curve: Curves.easeOutCubic),
    );
    _titleOpacity = Tween<double>(begin: 0, end: 1).animate(
      CurvedAnimation(
        parent: _splashController,
        curve: const Interval(0.24, 0.9, curve: Curves.easeInOut),
      ),
    );
    _signatureOpacity = Tween<double>(begin: 0, end: 1).animate(
      CurvedAnimation(
        parent: _splashController,
        curve: const Interval(0.58, 1, curve: Curves.easeIn),
      ),
    );
    _splashController.forward();
  }

  @override
  void dispose() {
    _splashController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.transparent,
      body: Stack(
        children: [
          AnimatedBuilder(
            animation: Listenable.merge([_logoScale, _logoAlignY]),
            builder: (context, child) => Align(
              alignment: Alignment(0, _logoAlignY.value),
              child: Transform.scale(
                scale: _logoScale.value,
                alignment: Alignment.center,
                child: child,
              ),
            ),
            child: const OpenGolfLogoBanner(width: 280),
          ),
          Align(
            alignment: Alignment.center,
            child: Padding(
              padding: const EdgeInsets.only(top: 246),
              child: FadeTransition(
                opacity: _titleOpacity,
                child: Text(
                  'GESTIONE\nSQUADRE',
                  textAlign: TextAlign.center,
                  style: GoogleFonts.blackOpsOne(
                    color: Colors.white,
                    fontSize: 44,
                    height: 1.04,
                    fontWeight: FontWeight.w400,
                    letterSpacing: 1.6,
                    shadows: const [
                      Shadow(color: Colors.black, blurRadius: 8),
                      Shadow(color: Colors.black, blurRadius: 8),
                    ],
                  ),
                ),
              ),
            ),
          ),
          Positioned(
            right: 16,
            bottom: 38,
            child: FadeTransition(
              opacity: _signatureOpacity,
              child: Container(
                padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 4),
                decoration: BoxDecoration(
                  color: Colors.black.withValues(alpha: 0.42),
                  borderRadius: BorderRadius.circular(8),
                ),
                child: const Text(
                  'by R. Ronco',
                  style: TextStyle(
                    color: Colors.white,
                    fontStyle: FontStyle.italic,
                    fontSize: 18,
                    fontWeight: FontWeight.w400,
                    shadows: [
                      Shadow(color: Colors.black, blurRadius: 8),
                    ],
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

File: app_mobile/kmp/androidApp/src/main/java/com/ansmi/gestsquadre/kmp/ui/LaunchSplashScreen.kt
package com.ansmi.gestsquadre.kmp.ui

import androidx.compose.animation.core.Animatable
import androidx.compose.animation.core.CubicBezierEasing
import androidx.compose.animation.core.tween
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.BiasAlignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.scale
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Shadow
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.Font
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.ansmi.gestsquadre.kmp.R
import com.ansmi.gestsquadre.kmp.ui.components.OpenGolfLogoBanner
import com.ansmi.gestsquadre.kmp.ui.theme.GlobalAppBackground

private val EaseOutCubic = CubicBezierEasing(0.33f, 1f, 0.68f, 1f)

private val BlackOpsOneFamily =
    FontFamily(Font(R.font.black_ops_one, FontWeight.Normal))

@Composable
fun LaunchSplashScreen() {
    val progress = remember { Animatable(0f) }

    LaunchedEffect(Unit) {
        progress.animateTo(
            targetValue = 1f,
            animationSpec = tween(durationMillis = 4300, easing = EaseOutCubic),
        )
    }

    val logoScale = 0.34f + (0.68f * progress.value)
    val logoAlignY = -0.74f + (0.36f * progress.value)
    val titleAlpha =
        when {
            progress.value < 0.24f -> 0f
            progress.value > 0.9f -> 1f
            else -> (progress.value - 0.24f) / (0.9f - 0.24f)
        }
    val signatureAlpha =
        when {
            progress.value < 0.58f -> 0f
            else -> (progress.value - 0.58f) / (1f - 0.58f)
        }

    Box(modifier = Modifier.fillMaxSize()) {
        GlobalAppBackground()
        Box(
            modifier = Modifier.fillMaxSize(),
            contentAlignment = BiasAlignment(0f, logoAlignY),
        ) {
            OpenGolfLogoBanner(
                modifier = Modifier.scale(logoScale),
                width = 286f,
            )
        }
        Text(
            text = "GESTIONE\nSQUADRE",
            modifier =
                Modifier
                    .align(Alignment.Center)
                    .padding(top = 246.dp)
                    .alpha(titleAlpha),
            textAlign = TextAlign.Center,
            fontFamily = BlackOpsOneFamily,
            style =
                TextStyle(
                    color = Color.White,
                    fontSize = 44.sp,
                    lineHeight = 46.sp,
                    fontWeight = FontWeight.Normal,
                    letterSpacing = 1.6.sp,
                    shadow = Shadow(color = Color.Black, blurRadius = 8f),
                ),
        )
        Surface(
            modifier =
                Modifier
                    .align(Alignment.BottomEnd)
                    .padding(end = 16.dp, bottom = 38.dp)
                    .alpha(signatureAlpha),
            color = Color.Black.copy(alpha = 0.42f),
            shape = RoundedCornerShape(8.dp),
        ) {
            Text(
                text = "by R. Ronco",
                modifier = Modifier.padding(horizontal = 10.dp, vertical = 4.dp),
                color = Color.White,
                fontSize = 18.sp,
                fontStyle = FontStyle.Italic,
                style = TextStyle(shadow = Shadow(color = Color.Black, blurRadius = 8f)),
            )
        }
    }
}
